frappe/desk/form/assign_to.py

```diff
@@ -93,8 +93,9 @@ def add(args=None):
 
 			doc = frappe.get_doc(args["doctype"], args["name"])
 
-			# if assignee does not have permissions, share
-			if not frappe.has_permission(doc=doc, user=assign_to):
+			# if sharing is not disabled & assignee does not have permissions, share
+			sharing_disabled = frappe.get_system_settings("disable_document_sharing")
+			if not sharing_disabled and not frappe.has_permission(doc=doc, user=assign_to):
```

Comment:
Let's at least tell the assigner if the assignee doesn't have permission? Improves UX. A message print or alert should be fine.

Reply:
@ankush done
<img width="1290" alt="Screenshot 2023-03-24 at 3 05 13 PM" src="https://user-images.githubusercontent.com/25857446/227483005-04a9aca1-9f22-4774-b56f-cd96baf31dac.png">

Reply:
Does it make sense to allow assignment if the user cannot access it? Or should we `throw` instead of `msgprint`? I could imagine that the assigned user might still get confused when they get a todo without access. Just a thought, no strong preference here.

Reply:
@marination throw is fine too. :+1: Anything that avoids future confusion.